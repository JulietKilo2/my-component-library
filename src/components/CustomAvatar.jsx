import React from "react";
import styles from "./CustomAvatar.module.css";

const CustomAvatar = ({ imgSrc, altText, customStyles }) => {
  return (
    <img
      className={styles.customAvatar}
      style={{ customStyles }}
      src={imgSrc || ""}
      alt={altText || ""}
    />
  );
};

export default CustomAvatar;
