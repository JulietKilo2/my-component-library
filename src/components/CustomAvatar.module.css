.customAvatar {
  border-radius: 50%;
}
