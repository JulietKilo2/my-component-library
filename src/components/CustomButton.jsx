import React from "react";
import styles from "./CustomButton.module.css";

const CustomButton = ({ children, customStyles, buttonEvent }) => {
  const handleOnClick = () => {
    buttonEvent && buttonEvent();
  };

  return (
    <button
      onClick={handleOnClick}
      className={styles.customButton}
      style={customStyles}
    >
      {children}
    </button>
  );
};

export default CustomButton;
