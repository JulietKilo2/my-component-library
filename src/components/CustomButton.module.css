.customButton {
  padding: 7px 14px;
  border: none;
  cursor: pointer;
}
