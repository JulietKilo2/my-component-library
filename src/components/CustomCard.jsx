import React from "react";
import styles from "./CustomCard.module.css";

const CustomCard = ({ children, customStyle }) => {
  return (
    <div className={styles.customCard} style={customStyle || {}}>
      {children}
    </div>
  );
};

export default CustomCard;
