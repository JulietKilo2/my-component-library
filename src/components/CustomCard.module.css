.customCard {
  width: fit-content;
  height: auto;
  border-radius: 8px;
  background-color: white;
  padding: 2rem 1rem;
  border: 1px solid white;
}
