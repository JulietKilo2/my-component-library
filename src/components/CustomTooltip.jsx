import React, { useState } from "react";
import styles from "./CustomTooltip.module.css";

const CustomTooltip = ({ children, label, customStyles }) => {
  const [tooltipToggle, setTooltipToggle] = useState(false);
  const onMouseEnter = () => setTooltipToggle(true);
  const onMouseLeave = () => setTooltipToggle(false);
  return (
    label && (
      <div className={styles.customTooltipOuterWrapper}>
        <div
          className={styles.customTooltipInnerWrapper}
          onMouseEnter={onMouseEnter}
          onMouseLeave={onMouseLeave}
        >
          {children}
          <span
            style={customStyles}
            className={
              tooltipToggle
                ? styles.customTooltipVisible
                : styles.customTooltipInvisible
            }
          >
            {label}
          </span>
        </div>
      </div>
    )
  );
};

export default CustomTooltip;
