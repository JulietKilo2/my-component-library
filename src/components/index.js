import CustomAvatar from "./CustomAvatar";
import CustomButton from "./CustomButton";
import CustomCard from "./CustomCard";
import CustomTooltip from "./CustomTooltip";

export { CustomAvatar, CustomButton, CustomCard, CustomTooltip };
