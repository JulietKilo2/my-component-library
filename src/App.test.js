import { render, screen, fireEvent, toBe } from "@testing-library/react";
import { toBeInTheDocument, toHaveBeenCalled } from "@testing-library/jest-dom";
import { CustomButton, CustomTooltip } from "./components";
import { Add } from "./utils/Add";

/**
 * JEST
 */

// test suite (테스트 묶음)
describe("true is truthy and false is falsy", () => {
  // test case (테스트 케이스)
  // test() or it()
  // both are same
  test("true is truthy", () => {
    // assertions - what you put into the test case
    const variable = "string";
    expect(typeof variable === "string").toBe(true);
  });

  it("math", () => {
    expect(Add(1, 2)).toBe(3);
    expect(Add(2, 3)).toBe(5);
  });
});

/**
 * REACT TESTING LIBRARY
 */

describe("test CustomTooltip", () => {
  test("render custom tooltip", () => {
    const utils = render(
      <CustomTooltip label={"tooltip"}>{"content"}</CustomTooltip>
    );
    // when assigning a render function to a variable, the name should be either "utils" or "view"
    expect(utils.container).toMatchSnapshot();
    // .toMatchSnapshot() creates a snapshot in the directory of __snapshots__ to check if the rendered results are the same after updating the component.
    // after saving the changes, press 'u' in the terminal to update the snapshot.
  });

  test("check if the desired text variables are rendered", () => {
    render(<CustomTooltip label={"tooltip"}>{"content"}</CustomTooltip>);
    expect(screen.getByText("content")).toBeInTheDocument();
  });

  test("check if the tooltip text shows up", () => {
    render(<CustomTooltip label={"tooltip"}>{"content"}</CustomTooltip>);
    const customToolTip = screen.queryByText("tooltip");
    fireEvent.click(customToolTip, () => {
      expect(screen.getByText("content")).toBeInTheDocument();
    });
  });
});

describe("test CustomButton", () => {
  test("render test button", () => {
    render(<CustomButton>{"button text"}</CustomButton>);
  });

  // test click event
  test("testing on click", () => {
    const newEvent = () => {
      const newParagraph = document.createElement("p");
      const newContent = document.createTextNode("Hello world");
      newParagraph.appendChild(newContent);
      document.body.append(newParagraph);
    };
    render(<CustomButton buttonEvent={newEvent}>{"button text"}</CustomButton>);
    const button = screen.getByRole("button");
    fireEvent.click(button, () => {
      expect(screen.getByText("Hello world")).toBeInTheDocument();
    });
    // expect(screen.getByText("Hello world")).toBeInTheDocument();
    // screen.debug();
  });
});
