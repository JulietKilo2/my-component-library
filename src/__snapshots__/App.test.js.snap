// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CustomTooltip render custom tooltip 1`] = `
<div>
  <div
    class="customTooltipOuterWrapper"
  >
    <div
      class="customTooltipInnerWrapper"
    >
      content
      <span
        class="customTooltipInvisible"
      >
        tooltip
      </span>
    </div>
  </div>
</div>
`;
